metric_providers/network/proxy/proxy_provider.py

```diff
@@ -0,0 +1,146 @@
+# pylint: disable=no-member,consider-using-with,subprocess-popen-preexec-fn,import-error,too-many-instance-attributes,too-many-arguments
+
+# This code handles the setup of the proxy we use to monitor the network connections in the docker containers.
+# Structurally it is a copy of the BaseMetricProvider but because we need to do things slightly different it is a copy.
+# In the future this might be implemented as a proper provider.
+
+import os
+from pathlib import Path
+import subprocess
+import signal
+import sys
+import time
+import re
+from datetime import datetime, timezone
+import platform
+import subprocess
+from packaging.version import parse
+
+from db import DB
+from global_config import GlobalConfig
+from metric_providers.base import MetricProviderConfigurationError
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+
+class NetworkConnectionsProxyContainerProvider:
+
+    def __init__(self, host_ip=None):
+        self._metric_name = 'dockerproxy'
```

Comment:
can we rename this to network_connections_proxy_container and then also reflect this in the `config.yml`